cogs/music_cog.py

```diff
@@ -84,32 +84,49 @@ async def command_play(self, ctx, *args):
 
     @commands.command(name="pause", help="Pauses the current song")
     async def pause(self, ctx, *args):
-        #TODO
-        print("todo")
+        if self.is_playing:
+            self.is_playing = False
+            self.is_paused = True
+            self.vc.pause()
+        elif self.is_paused:
+            self.vc.resume()
 
     @commands.command(name="resume", aliases=["r"], help="Resumes the current song")
     async def resume(self, ctx, *args):
-        #TODO
-        print("todo")
+        if self.is_paused:
+            self.is_paused = False
+            self.is_playing = True
+            self.vc.resume()
```

Comment:
Extract this into its own function. Then you can call that on line 92 as well so it will handle the is_paused and is_playing states as well